### PR TITLE
[BT-5129-logging-tracing] Enable debug / tracing for the JVM driver.

### PR DESCRIPTION
**Pull request by findgriffin**
### Description
Firstly I implemented request/response logging for the JVM driver. Secondly I think I fixed a bug where we were ignoring the `FAUNA_ENDPOINT` and `FAUNA_SECRET` environment variables.

The new debug facility can be enabled by setting `FAUNA_DEBUG=1` and disabled with `FAUNA_DEBUG=0` (or if the environment variable is missing). This will enable logging to `stderr` with `Leve.FINE` which is the equivalent of Log4j `DEBUG`. Users can also pass in their own log handler if they want to use a different log level, write to a file etc.

### Motivation and context

We recently had an event where a customer was receiving 401 Unauthorized responses from Fauna, and it was hard to tell why. If we had an easy way to see the headers they were sending on the client side, it would have been much easier to debug.

### How was the change tested?

Other than running the tests normally, you can see the comment in `FaunaConfigTest` to see how I tested the environment variables.

### Change types

* - [x] Bug fix (non-breaking change that fixes an issue)
* - [x] New feature (non-breaking change that adds functionality)
* - [ ] Breaking change (backwards-incompatible fix or feature)

### Checklist:

* - [x] My code follows the code style of this project.
* - [ ] My change requires a change to Fauna documentation.
* - [ ] My change requires a change to the README, and I have updated it accordingly.

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/fauna/client/FaunaClient.java [hunk 1]
```diff
@@ -86,17 +101,28 @@ private <T> void completeRequest(QuerySuccess<T> success, Throwable throwable) {
         }
     }
 
+    private void logResponse(HttpResponse<InputStream> response) {
+        logger.fine(MessageFormat.format("Fauna HTTP Response {0} from {1}, headers: {2}",
```
[the remark this answers is not in the document]
Comment (findgriffin) [reply src/main/java/com/fauna/client/FaunaClient.java:105]: If they pass their own logger, then our configuration gets ignored. Maybe I could just make this clearer in the documentation and Javadocs?

File: README.md [hunk 2]
```diff
@@ -580,3 +580,28 @@ public class App {
     }
 }
 ```
+
+## Debugging / Tracing
+
+If you would like to see the requests and responses the client is making and receiving, you can set the environment
+variable `FAUNA_DEBUG=DEBUG`, to see the request bodies you can set `FAUNA_DEBUG=TRACE`. You can also pass in your own
```
[the remark this answers is not in the document]
Comment (findgriffin) [reply README.md]: I made `FAUNA_DEBUG` binary.

File: src/main/java/com/fauna/client/FaunaClient.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (findgriffin) [reply src/main/java/com/fauna/client/FaunaClient.java:105]: I removed this code by making `FAUNA_DEBUG` binary.

File: src/main/java/com/fauna/client/Logging.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.fauna.client;
+
+import java.net.http.HttpHeaders;
+import java.util.stream.Collectors;
+
+public class Logging {
+    static String logHeaders(HttpHeaders headers) {
```
[the remark this answers is not in the document]
Comment (findgriffin) [reply src/main/java/com/fauna/client/Logging.java]: Renamed to `headersAsString`.

**Review by pnwpedro — APPROVED** [review]

**Pull request merged**